libusb/libusb.h

```diff
@@ -50,11 +50,18 @@ typedef SSIZE_T ssize_t;
 #include <time.h>
 
 #if defined(__STDC_VERSION__) && (__STDC_VERSION__ >= 199901L)
-#define ZERO_SIZED_ARRAY		/* [] - valid C99 code */
+#define LIBUSB_FLEXIBLE_ARRAY		/* [] - valid C99 code */
 #else
-#define ZERO_SIZED_ARRAY	0	/* [0] - non-standard, but usually working code */
+#define LIBUSB_FLEXIBLE_ARRAY	0	/* [0] - non-standard, but usually working code */
 #endif /* __STDC_VERSION__ */
 
+/* For when this is ready https://people.kernel.org/kees/bounded-flexible-arrays-in-c */
+#if defined(__STDC_VERSION__) && (__STDC_VERSION__ >= 199901L)
+#define LIBUSB_COUNTED_BY(x)
+#else
+#define LIBUSB_COUNTED_BY(x)
```

Comment:
is something missing here?

Reply:
Yes, see a couple of lines up. I can elaborate on the comment though. Missing is compiler support for this.

The idea was just to get ready for it. And in trying to figure out the length of these arrays, I think I found an overread. Hopefully you all can review and tell me if I'm right about it...

Reply:
Comment improved.